pkg/template/core/starlark_value.go

```diff
@@ -54,7 +53,11 @@ func (e StarlarkValue) AsInt64() (int64, error) {
 
 func (e StarlarkValue) asInterface(val starlark.Value) interface{} {
 	if obj, ok := val.(StarlarkValueToGoValueConversion); ok {
-		return obj.AsGoValue()
+		res, err := obj.AsGoValue()
+		if err != nil {
+			panic(err)
```

Comment:
Regardless of how you decide that question "`represent` vs. `StarlarkValueToGoValueConversion` vs. ....?" (summarized above), seems like we can provide a significantly better experience to the end user if we return an `error` rather than panic out of `StarlarkValue.asInterface()`.

We'd love to preserve the panic for situations where we truly need to abort rather than as an error handling technique.

`AsGoValue()` already returns an `error`, so I think most of the plumbing is in place from there.

Doing so means that the internal Go stack trace will just go away, leaving the user with a short (and easier to read) error message.